### ISSUE TITLE
form.resetDirty is unreliable when called with partial data

### ISSUE THREAD

**Issue by btmnk** [mantinedev/mantine#3896]
### What package has an issue

@mantine/form

### Describe the bug

When updating the form with an async response the useForm recipe currently shows 

```tsx
  useEffect(() => {
    loadInitialValues().then((values) => {
      form.setValues(values);
      form.resetDirty(values);
    });
  }, []);
```

This works fine in the assumption that `values` contains the very same structure as the form will have after `form.setValues(values)` has updated the state. That's not always the case though.
When calling`form.setValues(values)` the form will actually just overwrite the current values with the passed object.
If `values` is just a partial of the full form values than we still keep the not updated previous values.
 
For `form.resetDirty(values)` that's not the case. Instead the form will take the partial as the new snapshot and compare it with the merged values and return true again. This kindof forces us to make absolutely sure that we always pass the full structure into `resetDirty()` and fill up any missing values that might not always be part of an API response for example.

Right now I'm solving this by using `form.resetDirty({ ...defaults, ...data })` which seems to be enough for the time being.

I think it would already help if this was explained at https://mantine.dev/form/recipes/#set-initial-values-with-async-request

### What version of @mantine/hooks page do you have in package.json?

6.0.4

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/damp-http-x9y3qg?file=/src/App.tsx

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

Maybe we could pass the data to `form.reset(data)` so it uses that instead of the initialValues to reset the form so we can get a clean form state with the response. In this case we could also just merge the initialValues with the passed data so it keeps the defaults for unset properties.

**Comment by rtivital** [comment]
You are welcome to submit a pull request that will provide support for values partial in `form.resetDirty`. No need to update anything in the documentation: code itself and a test will be enough.

**Comment by btmnk** [comment]
I wonder if it would make more sense to extend the setValues function...

E.g. the actual thing we want to do usually is to just update the values of the form without marking it as dirty. Using two functions after each other who both update some internal state seems a bit counter intuitive to me. Especially considering that it's not transparent what would happen if one would just call `form.resetDirty` without setValues.

Wouldn't something like
```ts
setValues = ({ values: Values, resetDirty: boolean, reset: boolean }) => ...
```
be more useful?

**Comment by rtivital** [comment]
Currently, `setValues` does exactly what it says – it updates the state with given values, and I think it is fine. Any changes to it are a breaking change, it cannot be implemented, at least soon.

**Comment by rtivital** [comment]
Fixed in 6.0.5
